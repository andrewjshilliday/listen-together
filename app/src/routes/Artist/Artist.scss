.loading {
  height: 100%;
  justify-content: center;
  align-items: center;
  display: flex;
}

.artist {
  display: flex;
}

.sidebar {
  width: 30%;
  min-width: 250px;
  max-width: 400px;
  padding: 10px 15px;

  &.sticky {
    position: sticky;
    top: 0;
  }
  .editorial-notes {
    overflow: auto;
    font-size: 10pt;
  }
}

.track-list-container {
  width: 70%;
  padding-left: 15px;
  .track-list-header {
    position: sticky;
    top: 0;
    padding-bottom: 10px;
    background: white;
  }
}
