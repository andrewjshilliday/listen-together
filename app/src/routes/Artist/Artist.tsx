import React, { useState, useEffect } from 'react';
import { MusicKitApiService } from '../../services';
import { Loading, MediaItemCardCarousel } from '../../components/common';
import styled from 'styled-components';

const Artist: React.FC = (props: any) => {
  const id = props.match.params.id;
  const [loading, setLoading] = useState(true);
  const [artist, setArtist] = useState({} as MusicKit.MediaItem);
  const [albums, setAlbums] = useState([] as MusicKit.MediaItem[]);
  const [playlists, setPlaylists] = useState([] as MusicKit.MediaItem[]);

  useEffect(() => {
    const getSearchResults = async () => {
      setLoading(true);
      setArtist(await MusicKitApiService.Artist(id, 'albums,playlists'));
      setLoading(false);
    };
    getSearchResults();
  }, [id]);

  useEffect(() => {
    const getAlbumRelationships = async () => {
      if (artist.relationships && artist.relationships.albums) {
        setAlbums(artist.relationships.albums.data);
        /* setAlbums(await MusicKitApiService.Albums(artist.relationships.albums.data.map((i: MusicKit.MediauItem) => i.id))); */
      }
    };
    const getPlaylistRelationships = async () => {
      if (artist.relationships && artist.relationships.playlists) {
        setPlaylists(artist.relationships.playlists.data);
        /* setPlaylists(await MusicKitApiService.Playlists(artist.relationships.playlists.data.map((i: MusicKit.MediaItem) => i.id))); */
      }
    };
    getAlbumRelationships();
    getPlaylistRelationships();
  }, [artist]);

  if (loading) return (<Loading />);
  return (
    <ArtistContainer>
      <h1 className="text-truncate">{artist.attributes.name}</h1>
      {
        albums.length > 0 &&
        <>
          <h2>Albums</h2>
          <MediaItemCardCarousel items={albums}></MediaItemCardCarousel>
        </>
      }
      {
        playlists.length > 0 &&
        <>
          <h2>Playlists</h2>
          <MediaItemCardCarousel items={playlists}></MediaItemCardCarousel>
        </>
      }
    </ArtistContainer>
  );
}

export default Artist;


const ArtistContainer = styled.div``;
