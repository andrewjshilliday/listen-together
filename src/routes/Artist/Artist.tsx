import React, { useState, useEffect } from 'react';
import { MKArtist } from '../../services/MusicKitApi';
/* import { MediaItemCardCarousel } from '../../components/common'; */
import CircularProgress from '@material-ui/core/CircularProgress';
import './Artist.scss';

const Artist: React.FC = (props: any) => {
  const [artist, setArtist] = useState({} as any);
  const [loading, setLoading] = useState(true);
  const id = props.match.params.id;

  useEffect(() => {
    const getSearchResults = async () => {
      setLoading(true);
      setArtist(await MKArtist(id, 'relationships'));
      setLoading(false);
      const asd = await MKArtist(id);
      console.log(asd);
    };
    getSearchResults();
  }, [id]);

  if (loading) return (<div className="loading"><CircularProgress /></div>);
  
  return (
    <div className="artist view">
      <h1 className="text-truncate">{artist.attributes.name}</h1>
      {/* {
        artist.relationships && artist.relationships.albums &&
        <>
          <h2>Albums</h2>
          <MediaItemCardCarousel items={artist.relationships.albums.data}></MediaItemCardCarousel>
        </>
      } */}
    </div>
  );
}

export default Artist;
