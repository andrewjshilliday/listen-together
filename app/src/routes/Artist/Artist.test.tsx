import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import Artist from './Artist';

jest.mock('../../services/MusicKit');
jest.mock('../../services/MusicKitApi', () => ({
  MusicKitApiService: require('../../__mocks__/services/MusicKitApi/MusicKitApi.mock').default
}));
jest.mock('../../services/ListenTogetherApi', () => ({
  ListenTogetherApiService: require('../../__mocks__/services/ListenTogetherApi/ListenTogetherApi.mock').default
}));

jest.useFakeTimers();

it('renders correctly', async () => {  
  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/artist/266463936' ]}>
      <Route path="/artist/:id" component={Artist} />
    </MemoryRouter>);

  jest.runTimersToTime(1);

  await act(async () => {
    await Promise.resolve();
  });

  expect(tree.toJSON()).toMatchSnapshot();
});
