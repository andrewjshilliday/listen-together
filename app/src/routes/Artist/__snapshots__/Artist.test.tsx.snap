// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-fzoiQi jkvegY"
>
  <h1
    className="text-truncate"
  >
    Gary Clark Jr.
  </h1>
  <div
    className="sc-fzqARJ jmlzcM"
  >
    <img
      className="sc-fzqNqU eHNWdn"
    />
  </div>
  <div
    className="sc-AxirZ fJUMyY"
  >
    <h2>
      Albums
    </h2>
    <div
      className="sc-AxiKw enaPwj"
    >
      <div
        className="sc-AxhCb iEIFwb"
        onClick={[Function]}
      />
      <div
        className="sc-AxgMl eFueIV"
      >
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1447538299"
            onClick={[Function]}
          >
            <img
              alt="This Land"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              This Land
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1022109426"
            onClick={[Function]}
          >
            <img
              alt="The Story of Sonny Boy Slim"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              The Story of Sonny Boy Slim
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/906127510"
            onClick={[Function]}
          >
            <img
              alt="Gary Clark Jr. Live"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Gary Clark Jr. Live
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/560639480"
            onClick={[Function]}
          >
            <img
              alt="Blak and Blu"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Blak and Blu
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1200488004"
            onClick={[Function]}
          >
            <img
              alt="Live North America 2016"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Live North America 2016
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/560652745"
            onClick={[Function]}
          >
            <img
              alt="Blak and Blu (Deluxe Version)"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Blak and Blu (Deluxe Version)
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/626872587"
            onClick={[Function]}
          >
            <img
              alt="iTunes Session"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              iTunes Session
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
      </div>
      <div
        className="sc-AxhCb sc-AxhUy ixXRnl"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="sc-AxirZ fJUMyY"
  >
    <h2>
      EPs & Singles
    </h2>
    <div
      className="sc-AxiKw enaPwj"
    >
      <div
        className="sc-AxhCb iEIFwb"
        onClick={[Function]}
      />
      <div
        className="sc-AxgMl eFueIV"
      >
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/455141519"
            onClick={[Function]}
          >
            <img
              alt="The Bright Lights - EP"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              The Bright Lights - EP
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1494805000"
            onClick={[Function]}
          >
            <img
              alt="This Land (Remix) [From the Tonight Show Starring Jimmy Fallon] [feat. Black Thought] - Single"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              This Land (Remix) [From the Tonight Show Starring Jimmy Fallon] [feat. Black Thought] - Single
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr. & The Roots
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1490025532"
            onClick={[Function]}
          >
            <img
              alt="Pearl Cadillac (feat. Andra Day) - Single"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Pearl Cadillac (feat. Andra Day) - Single
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1447106653"
            onClick={[Function]}
          >
            <img
              alt="This Land - Single"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              This Land - Single
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1279718975"
            onClick={[Function]}
          >
            <img
              alt="Come Together - Single"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Come Together - Single
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr. & Junkie XL
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1148774607"
            onClick={[Function]}
          >
            <img
              alt="Take Me Down - Single"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Take Me Down - Single
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1129970027"
            onClick={[Function]}
          >
            <img
              alt="Jam in the Van - Gary Clark Jr. - EP"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Jam in the Van - Gary Clark Jr. - EP
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr. & Jam in the Van
          </span>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/562514719"
            onClick={[Function]}
          >
            <img
              alt="Ain't Messin 'Round - Single"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Ain't Messin 'Round - Single
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Gary Clark Jr.
          </span>
        </div>
      </div>
      <div
        className="sc-AxhCb sc-AxhUy ixXRnl"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="sc-AxirZ fJUMyY"
  >
    <h2>
      Appears On
    </h2>
    <div
      className="sc-AxiKw enaPwj"
    >
      <div
        className="sc-AxhCb iEIFwb"
        onClick={[Function]}
      />
      <div
        className="sc-AxgMl eFueIV"
      >
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1447538299"
            onClick={[Function]}
          >
            <img
              alt="This Land"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              This Land
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/906127510"
            onClick={[Function]}
          >
            <img
              alt="Gary Clark Jr. Live"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Gary Clark Jr. Live
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1022109426"
            onClick={[Function]}
          >
            <img
              alt="The Story of Sonny Boy Slim"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              The Story of Sonny Boy Slim
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/560639480"
            onClick={[Function]}
          >
            <img
              alt="Blak and Blu"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Blak and Blu
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1200488004"
            onClick={[Function]}
          >
            <img
              alt="Live North America 2016"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Live North America 2016
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/455141519"
            onClick={[Function]}
          >
            <img
              alt="The Bright Lights - EP"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              The Bright Lights - EP
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/560652745"
            onClick={[Function]}
          >
            <img
              alt="Blak and Blu (Deluxe Version)"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Blak and Blu (Deluxe Version)
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/626872587"
            onClick={[Function]}
          >
            <img
              alt="iTunes Session"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              iTunes Session
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1279718975"
            onClick={[Function]}
          >
            <img
              alt="Come Together - Single"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Come Together - Single
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr. & Junkie XL
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1494805000"
            onClick={[Function]}
          >
            <img
              alt="This Land (Remix) [From the Tonight Show Starring Jimmy Fallon] [feat. Black Thought] - Single"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              This Land (Remix) [From the Tonight Show Starring Jimmy Fallon] [feat. Black Thought] - Single
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr. & The Roots
            </span>
          </a>
        </div>
      </div>
      <div
        className="sc-AxhCb sc-AxhUy ixXRnl"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="sc-AxirZ fJUMyY"
  >
    <h2>
      Playlists
    </h2>
    <div
      className="sc-AxiKw enaPwj"
    >
      <div
        className="sc-AxhCb iEIFwb"
        onClick={[Function]}
      />
      <div
        className="sc-AxgMl eFueIV"
      >
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/playlist/pl.a5ef67f3dde74a568e583ac9f0406d50"
            onClick={[Function]}
          >
            <img
              alt="Gary Clark Jr. Essentials"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Gary Clark Jr. Essentials
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Apple Music Blues
          </span>
        </div>
      </div>
      <div
        className="sc-AxhCb sc-AxhUy ixXRnl"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
